backend/src/api/controllers/career-path.controller.ts

```diff
@@ -7,23 +7,69 @@ import {
   getDomainLevels,
 } from '~/services/domain-level.service';
 import { upsertSkills } from '~/services/skill.service';
-import { upsertObjectives } from '~/services/skill-objective.service';
-import { createSkillCategories } from '~/services/skill-category.service';
+import {
+  getObjectives,
+  upsertObjectives,
+} from '~/services/skill-objective.service';
+import {
+  createSkillCategories,
+  getCategories,
+} from '~/services/skill-category.service';
 
 import { asyncForEach } from '~/common/helpers/array.helper';
 
+import { SkillObjective } from '~/data/entities/skill-objective';
 import { DomainLevel } from '~/data/entities/domain-level';
 import { Domain } from '~/data/entities/domain';
+import { Skill } from '~/data/entities/skill';
+
+type DomainLevelResponse = DomainLevel & {
+  skills: Skill[];
+};
 
-type CareeerPathType = {
+type CareeerPathResponse = {
   domain: Domain;
-  level: DomainLevel;
+  level: DomainLevelResponse;
+};
+
+const getLevelsAndSkills = async (
+  domain: Domain,
+): Promise<DomainLevelResponse> => {
+  const rootLevel = (await getDomainLevels(domain)) as DomainLevelResponse;
```

Comment:
getDomainLevels returns type `DomainLevel`. Do not use `as`, return correct format and use actual interface everywhere